docs: migrate from postgresql (#1257)
GreptimeTeam/docs: docs/user-guide/migrate-to-greptimedb/migrate-from-postgresql.md

right data types and "ttl" or "compaction" options, etc.

### Write data to both GreptimeDB and PostgreSQL simultaneously

Writing data to both GreptimeDB and PostgreSQL simultaneously is a practical strategy to avoid data loss during
migration. By utilizing PostgreSQL's client libraries (JDBC + a PostgreSQL driver), you can set up two client
instances - one for GreptimeDB and another for PostgreSQL. For guidance on writing data to GreptimeDB using SQL, please
refer to the [Ingest Data](/user-guide/ingest-data/for-iot/sql.md) section.

If retaining all historical data isn't necessary, you can simultaneously write data to both GreptimeDB and PostgreSQL
for a specific period to accumulate the required recent data. Subsequently, cease writing to PostgreSQL and continue
exclusively with GreptimeDB. If a complete migration of all historical data is needed, please proceed with the following
steps.

### Export data from PostgreSQL

[pg_dump](https://www.postgresql.org/docs/current/app-pgdump.html) is a commonly used tool to export data from
PostgreSQL. Using it, we can export the data that can be later imported into GreptimeDB directly. For example, if we

> want to export two schemas, `db1` and `db2` in the database `postgres` from PostgreSQL, we can use the following

Reviewer: ```suggestion
want to export schemas whose names start with `db` in the database `postgres` from PostgreSQL, we can use the following
```

command:

```bash
pg_dump -h127.0.0.1 -p5432 -Upostgres -ax --column-inserts --no-comments -n 'db*' postgres | grep -v "^SE" > /path/to/output.sql
```

Replace the `-h`, `-p` and `-U` flags with the appropriate values for your PostgreSQL server. The `-n` flag is used to

> specify the schemas to be exported. Add the database in the end. Note that we pipe the pg_dump output through a special

Reviewer: ```suggestion
specify the schemas to be exported. And the database `postgres` is at the end of the `pg_dump` command line. Note that we pipe the `pg_dump` output through a special
```

> `grep`, to remove some unnecessary "`SET`" or "`SELECT`" lines. The final output will be written to the

Reviewer: ```suggestion
`grep`, to remove some unnecessary `SET` or `SELECT` lines. The final output will be written to the
```

~ ❯ cat /path/to/output.sql

--
-- PostgreSQL database dump
--

-- Dumped from database version 16.4 (Debian 16.4-1.pgdg120+1)
-- Dumped by pg_dump version 16.4

--
-- Data for Name: foo; Type: TABLE DATA; Schema: db1; Owner: postgres
--

INSERT INTO db1.foo (ts, a) VALUES ('2024-10-31 00:00:00', 1);
INSERT INTO db1.foo (ts, a) VALUES ('2024-10-31 00:00:01', 2);
INSERT INTO db1.foo (ts, a) VALUES ('2024-10-31 00:00:01', 3);
INSERT INTO ...

--
-- Data for Name: foo; Type: TABLE DATA; Schema: db2; Owner: postgres
--

INSERT INTO db2.foo (ts, b) VALUES ('2024-10-31 00:00:00', '1');
INSERT INTO db2.foo (ts, b) VALUES ('2024-10-31 00:00:01', '2');
INSERT INTO db2.foo (ts, b) VALUES ('2024-10-31 00:00:01', '3');
INSERT INTO ...

--
-- PostgreSQL database dump complete
--
```

### Import data into GreptimeDB

The [psql -- PostgreSQL interactive terminal](https://www.postgresql.org/docs/current/app-psql.html) can be used to
import data into GreptimeDB. Continuing the above example, say the data is exported to file `/path/to/output.sql`, then
we can use the following command to import the data into GreptimeDB:

```bash
psql -h127.0.0.1 -p4003 -d public -f /path/to/output.sql
```

> Replace the `-h` and `-p` flags with the appropriate values for your GreptimeDB server. The `-d` flag is the database to
> use for psql. For psql to connecting to GreptimeDB, it should be fixed to "public". Add `-a` to the end to see every

Reviewer: ```suggestion
Replace the `-h` and `-p` flags with the appropriate values for your GreptimeDB server. The `-d` of the psql command is used to specify the database and cannot be omitted. The value `public` of `-d` is the default database used by GreptimeDB. You can also add `-a` to the end to see every
```
